### PR TITLE
WIP: Implement credits flow

### PR DESCRIPTION
**Pull request by essen**
I have manually verified that Cowboy stops reading from the socket
when credit flow blocks the process.

WIP because this depends on Cowboy 2.5 which isn't out yet and
will need to be updated in all relevant RabbitMQ projects first.

https://github.com/rabbitmq/rabbitmq-web-stomp/issues/40

### CODE REVIEW

File: src/rabbit_ws_client.erl [hunk 1]
```diff
@@ -26,7 +27,7 @@
 -export([init/1, handle_call/3, handle_info/2, terminate/2,
          code_change/3, handle_cast/2]).
 
--record(state, {conn, proc_state, parse_state, stats_timer, connection, heartbeat_mode, heartbeat, heartbeat_sup}).
+-record(state, {conn, proc_state, parse_state, state, conserve_resources, stats_timer, connection, heartbeat_mode, heartbeat, heartbeat_sup}).
```
Comment (michaelklishin) [inline src/rabbit_ws_client.erl]: This is node-local state, so presumably it is safe to go into 3.7.9, correct?

Comment (essen) [reply src/rabbit_ws_client.erl]: Node and process-local. It should not be an issue as long as live code reloading is not involved.

**Comment by essen** [comment]
This is missing the changes in emit_stats.

**Comment by essen** [comment]
The flow control state is now propagated properly to management UI.

**Comment by essen** [comment]
Travis fails because of an Elixir version check.

**Comment by michaelklishin** [comment]
@essen is this still relevant since #96, as in, should this PR be closed and a new one opened on top of what we have in #96?

**Comment by essen** [comment]
Right this will need to be rewritten, I'll do that on Tuesday (and should release Cowboy 2.6 on Monday so this will be a good time to update and get this change merged).

**Pull request closed without merging**